code/modules/research/techweb/_techweb_node.dm

```diff
@@ -83,10 +105,14 @@
 	if(host)
 		var/list/actual_costs = research_costs
 		if(host.boosted_nodes[id])
-			var/list/L = host.boosted_nodes[id]
-			for(var/i in L)
-				if(actual_costs[i])
-					actual_costs[i] -= L[i]
+			var/list/boostlist = host.boosted_nodes[id]
+			for(var/booster in boostlist)
+				if(actual_costs[booster])
+					actual_costs[booster] -= boostlist[booster]
+		for(var/actual_inc in actual_costs)
+			for(var/experi_type in discount_experiments)
```

Comment:
Better names (I don't know what the first one is, and expand experi to experiment).